src/main/java/frc/robot/subsystems/tank/TankSubsystem.java

```diff
@@ -20,6 +30,14 @@
   private final WPI_TalonSRX rightMain;
   private final WPI_TalonSRX rightFollow;
 
+  private final AHRS ahrs;
+
+  private final DifferentialDrivePoseEstimator odometry;
```

Comment:
in another comment I mentioned "all localization stuff"... we should take all the localization logic and put it in its own class and run it on its own thread. logically and from my experience, dead reckoning is more accurate the less time each loop is -- no matter your odometry method (e.g. circular), you're making an assumption about what happens between each tick, so reducing the tick interval reduces the fudginess of your guess. 

if we use the periodic() method, the commandbased system actively waits until an interval is hit before calling periodics, it's something like 20ms by default? i can't remember

Reply:
I second the localization system approach.